### PR TITLE
Fix for issue #16.

### PR DESCRIPTION
**Pull request by gvandegrift**
This PR allows for the consumer to configure the plugin to throw when the shell command returns a non-zero exit code (hence causing npm to fail with a non-zero exit code).

I am having problems with "npm run build".  It fails with the following error:

```
It looks like your Babel configuration specifies a module transformer. Please disable it. If you're using the "es2015" preset, consider using "es2015-rollup" instead. See https://github.com/rollup/rollup-plugin-babel#configuring-babel for more information
Error: It looks like your Babel configuration specifies a module transformer. Please disable it. If you're using the "es2015" preset, consider using "es2015-rollup" instead. See https://github.com/rollup/rollup-plugin-babel#configuring-babel for more information
    at preflightCheck (C:\src\webpack-shell-plugin\node_modules\rollup-plugin-babel\dist\rollup-plugin-babel.cjs.js:43:102)
    at transform$1 (C:\src\webpack-shell-plugin\node_modules\rollup-plugin-babel\dist\rollup-plugin-babel.cjs.js:104:18)
    at C:\src\webpack-shell-plugin\node_modules\rollup\www\ROLLUP\rollup\src\utils\transform.js:18:28
    at tryCatch (C:\src\webpack-shell-plugin\node_modules\rollup\www\ROLLUP\rollup\node_modules\es6-promise\lib\es6-promise\-internal.js:185:12)
    at invokeCallback (C:\src\webpack-shell-plugin\node_modules\rollup\www\ROLLUP\rollup\node_modules\es6-promise\lib\es6-promise\-internal.js:197:13)
    at C:\src\webpack-shell-plugin\node_modules\rollup\www\ROLLUP\rollup\node_modules\es6-promise\lib\es6-promise\then.js:25:7
    at flush (C:\src\webpack-shell-plugin\node_modules\rollup\www\ROLLUP\rollup\node_modules\es6-promise\lib\es6-promise\asap.js:87:5)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
    at Function.Module.runMain (module.js:577:11)
```

The only way I was able to get it to run to success was by removing .babelrc.  As a result, my index.js looks significantly different.  Can you advise on why I would be having the above failure?

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -62,65 +41,70 @@ function validateInput(options) {
...
-          _this.options.onBuildStart.forEach(function (script) {
+  apply(compiler) {
+
+    compiler.plugin('compilation', compilation => {
+      if (this.options.verbose) {
+        console.log(`Report compilation: ${ compilation }`);
+      }
+      if (this.options.onBuildStart.length) {
+        console.log('Executing pre-build scripts');
+        throw this.options.onBuildStart.forEach(script => {
```
Comment (opiepj) [inline lib/index.js]: I'm getting an undefined here when testing the code.

Comment (gvandegrift) [inline lib/index.js]: I ran `npm list -g` and don't see anything with "babel" in the name.  Is that what you meant by "installed globally"?

What version of npm and node are you running.  I am running node 6.2.0 and npm 3.8.9.  Perhaps that explains the difference?

**Comment by gvandegrift** [comment]
I was distracted by the rollup/es2015 issue, so I missed your point about the undefined.  I somehow got an extraneous throw inserted in that branch of the code.  I have removed it and pushed the commit.  It should work now.

I do still however have the problem with running `npm run build`.  There are two ways I can work around it.

- Remove .babelrc
- Follow these instructions:  https://github.com/rollup/rollup-plugin-babel/issues/72#issuecomment-238568551

The index.js that is in my PR was generated by just removing .babelrc.

**Comment by gvandegrift** [comment]
Do you know when you will be releasing a version of the plugin that has this code added?

**Pull request merged**